src/lsm/grid.zig

```diff
@@ -490,6 +490,22 @@ pub fn GridType(comptime Storage: type) type {
             completed_write.callback(completed_write);
         }
 
+        /// This function allows the caller to synchronously search for a block in the cache.
+        /// read_block encapsulates this logic but does so asynchronously with queuing and callback
+        /// resolution overheads.
+        pub fn read_from_cache(grid: *Grid, address: u64, checksum: u128) ?BlockPtrConst {
+            if (grid.cache.get_index(address)) |cache_index| {
+                const cache_block = grid.cache_blocks[cache_index];
+
+                const header = mem.bytesAsValue(vsr.Header, cache_block[0..@sizeOf(vsr.Header)]);
+                assert(header.op == address);
+                assert(header.checksum == checksum);
```

Comment:
I compared the performance with & without the asserts (there are more now). The numbers with & without the gated asserts are very similar. 
https://github.com/tigerbeetle/tigerbeetle/blob/00d9d2c3a88ecbfff3c0f02ec11203b5790c1c51/src/lsm/grid.zig#L583-L588

Resolving this conversation since the asserts don't seem to be imposing a performance penalty.